### PR TITLE
Upgrade to TS 3.0 and enable strict compiler options

### PR DESCRIPTION
**Pull request by flut1**
TS 3.0 brings little breaking changes so this is a relatively safe upgrade.

Enabling strict compiler options will result in more stable code for our modules.

### CODE REVIEW

**Comment by flut1** [comment]
@ThaNarie I changed the output directory of the definition files. Can you verify that this is an ok change?

File: tsconfig.build.json [hunk 1]
```diff
@@ -4,6 +4,6 @@
     "allowJs": false,
     "outDir": "tmp",
     "declaration": true,
-    "declarationDir": "./"
+    "declarationDir": "./declarations"
```
Comment (ThaNarie) [inline tsconfig.build.json]: Is there a specific reason for this? I liked having the declarations next to the compiled files, so it's easier to navigate to the .js file when ctrl+clicking on a file in your IDE (that will end up at the declaration).

Comment (flut1) [reply tsconfig.build.json]: Yeah, the TS compiler now seems to add the directory specified in declarationDir to the "exclude" option, resulting in no source files being picked up by the compiler. I can try to find a way around this. Do you know of any?

Comment (flut1) [reply tsconfig.build.json]: Nevermind, fixed it by adding an explicit `"exclude": []`

Comment (ThaNarie) [reply tsconfig.build.json]: We could output the declarations in `./tmp/declarations`, and them move them to the root from there after compilation?

**Comment by flut1** [comment]
Closing in favor of #60

**Pull request closed without merging**